clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesCachingAgent.java

```diff
@@ -72,31 +79,87 @@
 
   @Getter protected final Long agentInterval;
 
+  protected final KubernetesConfigurationProperties configurationProperties;
+
+  protected final KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap;
+
   protected KubernetesCachingAgent(
       KubernetesNamedAccountCredentials namedAccountCredentials,
       ObjectMapper objectMapper,
       Registry registry,
       int agentIndex,
       int agentCount,
-      Long agentInterval) {
+      Long agentInterval,
+      KubernetesConfigurationProperties configurationProperties,
+      KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap) {
     this.accountName = namedAccountCredentials.getName();
     this.credentials = namedAccountCredentials.getCredentials();
     this.objectMapper = objectMapper;
     this.registry = registry;
     this.agentIndex = agentIndex;
     this.agentCount = agentCount;
     this.agentInterval = agentInterval;
+    this.configurationProperties = configurationProperties;
+    this.kubernetesSpinnakerKindMap = kubernetesSpinnakerKindMap;
   }
 
   protected Map<String, Object> defaultIntrospectionDetails() {
     Map<String, Object> result = new HashMap<>();
     result.put("namespaces", getNamespaces());
-    result.put("kinds", primaryKinds());
+    result.put("kinds", kindsToCache());
     return result;
   }
 
   protected abstract List<KubernetesKind> primaryKinds();
 
+  protected abstract boolean cachesKind(KubernetesKind kind);
```

Comment:
Could use some javadoc here explaining why we need this method and what it's supposed to do.